### PR TITLE
UI test stubs on ostree feature

### PR DESCRIPTION
**Pull request by sghai**
Basic tests around ostree.

### CODE REVIEW

**Comment by oshtaier** [comment]
Just a suggestion. Usually we separate custom and RH content into different test cases. Please, do that if it is applicable for your situation

**Comment by sghai** [comment]
@oshtaier : I've created separate tests to cover custom and RH contents. Here is the list of combinations.

- custom ostree contents
- custom + RH contents
- custom ostree + other custom contents(yum + puppet)
- RH ostree + other RH contents (Yum)

Please point the tests which you feel should be defined separately based on contents.

**Comment by oshtaier** [comment]
For example, first one:
```
     @run_only_on('sat')
     @stubbed()
     @tier2
     def test_positive_add_custom_rh_ostree(self):
         """Create a CV with custom and RH ostree contents
 
         @Feature: Content View
 
         @Assert: CV should be created successfully with ostree contents
 
         @Status: Manual
         """
```

I mean something like that:
```
    @run_only_on('sat')
    @stubbed()
    @tier2
    def test_positive_add_custom_ostree(self):
        """Create a CV with custom ostree content
         @Feature: Content View
 
        @Assert: CV should be created successfully with ostree content
  
        @Status: Manual
        """
and
    @run_only_on('sat')
    @stubbed()
    @tier2
    def test_positive_add_rh_ostree(self):
        """Create a CV with RH ostree content
         @Feature: Content View
 
        @Assert: CV should be created successfully with ostree content
  
        @Status: Manual
        """
```
Sorry, if I missed the point completely, but I am referring to existing CV test cases

**Comment by sghai** [comment]
@oshtaier : thanks.. will separate them.

File: tests/foreman/ui/test_repository.py [hunk 1]
```diff
@@ -457,3 +457,44 @@ def test_positive_sync_custom_repo_docker(self):
...
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_custom_ostree_repo(self):
+        """Update custom ostree repository.
+
+        @Feature: Custom ostree repository
+
+        @Step: Update repo name and URL
```
Comment (sthirugn) [inline tests/foreman/ui/test_repository.py]: This is `Steps`. Testimony failed in travis for this reason :)

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -1370,3 +1370,170 @@ def test_positive_delete_version_with_ak(self):
...
+
+        @Assert: CV should be published with OStree contents
+
+        @Status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier2
+    def test_positive_promote_with_custom_ostree(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_contentview.py]: Can you please also add a test to undo a promotion? (remove the promoted content from an environment)

File: tests/foreman/ui/test_contentview.py [hunk 3]
```diff
@@ -1370,3 +1370,170 @@ def test_positive_delete_version_with_ak(self):
...
+        @Assert: CV should be published and promoted with custom OStree and all
+        other contents
+
+        @Status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier2
+    def test_positive_publish_with_rh_ostree(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_contentview.py]: Can you also add a test to remove a published version?

File: tests/foreman/ui/test_sync.py [hunk 4]
```diff
@@ -108,3 +108,33 @@ def test_positive_sync_disconnected_to_connected_rh_repos(self):
...
+    @run_only_on('sat')
+    @stubbed()
+    @tier2
+    def test_positive_sync_rh_ostree_repo(self):
+        """Sync CDN based ostree repository .
+
+        @Feature: RH ostree repository sync
+
+        @Steps:
+        1. Import a vlaid manifest
```
Comment (sthirugn) [inline tests/foreman/ui/test_sync.py]: typo: valid

**Comment by sthirugn** [comment]
Pending comments

File: tests/foreman/ui/test_contentview.py (hunk 3 above)
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: :+1:

File: tests/foreman/ui/test_contentview.py (hunk 2 above)
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: +1 thank you. completely missed.

**Comment by sghai** [comment]
All comments have been addressed and added new tests as per suggestion. Please review it again.. thank you.

**Comment by sthirugn** [comment]
ACK

**Comment by oshtaier** [comment]
When do we plan to have that functionality dev completed/implemented?

**Comment by oshtaier** [comment]
ACK

**Comment by sghai** [comment]
We are waiting on pulp 2.8 merge in upstream nightly.. should be done by late this week

**Comment by oshtaier** [comment]
aha, thanks @sghai for answer

**Pull request merged**